models/users/snowplow_web_users.sql

```diff
@@ -6,13 +6,17 @@
     disable_upsert_lookback=true,
     sort='start_tstamp',
     dist='domain_userid',
-    partition_by = {
+    partition_by = snowplow_utils.get_partition_by(bigquery_partition_by={
       "field": "start_tstamp",
       "data_type": "timestamp"
-    },
+    }),
```

Comment:
So Databricks handles partitions and timestamp columns differently to Snowflake/BQ. Databricks partitions based on each individual timestamp rather than on the date the timestamp is coming from, and there's no config we can use to change that. Instead, we'd have to introduce a new column and then partition based on that. Instead of doing that, I've decided to include tableProperties where databricks auto-optimises and partitions the dataset instead

Reply:
I'm not sure if I really like the tblproperties route and this way we could actually partition the data, this would be a breaking change if we add it later. We're already doing something similar in the `atomic.events` table to ensure that we can partition there

Reply:
it is hard to tell which version is better for customers with limited test data but it may be the case that both would bring benefits. As discussed, we could go with the custom partitioning first and if we get the feedback that it is not performant enough then add the autoOptimize bits.

Reply:
I've added the custom partitioning, I'll squash everything if it looks good to you!